src/copy.c

```diff
@@ -45,16 +45,77 @@
 #include "nodes/chunk_insert_state.h"
 #include "subspace_store.h"
 
+// PG14_GE is defined in compat/compat.h, so it must be included first
+#if PG14_GE
+#include <commands/copyfrom_internal.h>
+#endif
+
 /*
  * Copy from a file to a hypertable.
  *
  * Unfortunately, there aren't any good hooks in the regular COPY code to insert
- * our chunk dispatching. so most of this code is a straight-up copy of the
- * regular PostgreSQL source code for the COPY command (command/copy.c), albeit
- * with minor modifications.
+ * our chunk dispatching. So, most of this code is a straight-up copy of the
+ * regular PostgreSQL source code for the COPY command (command/copy.c
+ * and command/copyfrom.c), albeit with minor modifications.
  *
  */
 
+/*
+ * No more than this many tuples per CopyMultiInsertBuffer
+ *
+ * Caution: Don't make this too big, as we could end up with this many
+ * CopyMultiInsertBuffer items stored in CopyMultiInsertInfo's
+ * multiInsertBuffers list.  Increasing this can cause quadratic growth in
+ * memory requirements during copies into partitioned tables with a large
```

Comment:
You mean, there's a buffer per chunk, so it's multiplied, and big tables are likely to have more chunks, so it becomes kinda quadratic?

Reply:
Yes, exactly. This [comment](https://github.com/postgres/postgres/blob/6e2924b57793d662aa48bc88ac9e97d5d8eb90cf/src/backend/commands/copyfrom.c#L61) is a direct backport of the PG 14 code. 

I also thought about the comment and considered making it more precise. However, to make future backports as simple as possible, I changed as few as possible in the original code and comments.